### PR TITLE
feat(hook): model-routing tier-mismatch advisory

### PR DESCRIPTION
**Pull request by devseunggwan**
## 배경

user-level `CLAUDE.md`(= `ai-dotfiles/shared/AGENTS.md` 심링크, always-loaded)를 슬림화하는 작업의 **첫 단추**. prose 규칙 블록을 praxis 훅(enforcement) + 훅 `spec.md`(온디맨드 참조 지식)로 치환하는 패턴의 첫 적용 대상이 라우팅 블록이다.

라우팅 시그널은 이미 키워드→tier 매핑(`find→haiku`, `implement→sonnet`, `architect→opus`)이라 자동화(훅)의 일이지 지식 주입(스킬)이 아니다.

## 변경

- **신규 훅 `model-routing-advisory`** — PreToolUse(Bash) advisory-nudge (fail-open, exit 0, 차단 없음). 위임 명령의 `--model <tier>`가 task 키워드가 암시하는 tier와 불일치하면 stderr nudge. dispatched Bash 그룹 멤버(개별 `.sh` 없음).
- **훅 `spec.md`에 Skill & Agent Routing 결정 트리 전문** 수록 — 참조 SoT, always-loaded 아님.
- 등록 5면: `manifest.json` / `spec.md`(`Supported hosts: claude`) / `docs/hook/INDEX.md` / `ARCHITECTURE.md` hook index / `docs/bypass-vars.md` + 빌드 재생성(stub·operating-matrix).
- 검출은 raw-string 스캔(중첩 `--command "... --model ..."` 대응); 키워드 경계 `[a-z0-9_-]` 제외로 `--add-dir` 등 flag/경로 토큰 hijack 방지.

## 설계 결정 / 한계 (spec.md 문서화)

- **v1 = Bash 위임만**. Task/Agent는 tier가 `subagent_type` 암시(model은 optional override)라 비교 대상 부재 → 범위 밖(follow-up).
- **regex 시그널 미모델링**: `debug.*race`/`system.*design`은 리터럴-키워드 훅으로 표현 불가 → bare `debug`는 opus 신호 아님(문서화된 false-negative).
- **negation/status 문맥 미파싱**, **prompt 내 `--model` 리터럴 first-match 오독**: anti-fragile(NLP/구조 파서 회피), advisory+opt-out으로 흡수.

## 검증

- 훅 테스트 37건 전부 통과(mismatch surface / match·비교불가·경계충돌·trailing-punct·negation-pin·opt-out·fail-open).
- 독립 리뷰 2종: `oh-my-claudecode:code-reviewer`(MEDIUM 1 — flag-token hijack 수정) + codex(P2 3 — `opus;` 파싱·bare debug 승격·negation 각각 수정/revert/문서화).

Pre-commit verified: `python3 -m pytest -q` → 469 passed; `bash tests/hooks/advisory-nudge/test_model_routing_advisory.sh` → 37 passed; `ruff check` clean; `python3 scripts/check-plugin-manifests.py` → OK
Caller chain verified: `_dispatch.group_members("PreToolUse","Bash")`가 `manifest.json`에서 자동 열거(별도 registry 없음) → `test_dispatch.py`가 exact-Bash 35 / advisory-nudge 13 / codex 31(claude-only 제외) 단언, `build-plugin-manifests.py` 재생성 후 working-tree drift 0

Closes #786

## Summary by CodeRabbit

* **New Features**
  * Added an advisory when delegated Bash tasks use a model tier that may not match the task’s complexity.
  * Advisories identify whether the selected model appears over- or under-powered.
  * Added opt-out support through an environment variable or inline acknowledgement marker.
* **Documentation**
  * Added guidance, operating details, routing examples, and troubleshooting information for the new advisory.
* **Tests**
  * Added coverage for routing matches, mismatches, bypasses, malformed input, and edge cases.

### CODE REVIEW

File: docs/hook/INDEX.md [hunk 1]
```diff
@@ -77,6 +77,7 @@ Fail-open on infrastructure errors by design.
...
 | [bash-worktree-existence-advisory](../../hooks/advisory-nudge/bash-worktree-existence-advisory/spec.md) | PreToolUse | Advisory nudge when `cd <path>` targets a path that does not exist on disk |
 | [pre-commit-staged-file-enumeration](../../hooks/advisory-nudge/pre-commit-staged-file-enumeration/spec.md) | PreToolUse | Advisory nudge before `git commit` listing staged file additions not created via Write/Edit/MultiEdit/NotebookEdit this session (heredoc / `> file` / external-script output caught before it rides into the commit) — issue #784 |
+| [model-routing-advisory](../../hooks/advisory-nudge/model-routing-advisory/spec.md) | PreToolUse | Advisory nudge when a Bash delegation's `--model <tier>` mismatches the tier implied by task keywords (`find → haiku`, `implement → sonnet`, `architect/security → opus`); replaces the always-loaded `Skill & Agent Routing` + `Model Routing Rules` blocks, whose full tree now lives in this hook's spec — issue #786 |
```
[the remark this answers is not in the document]
Comment (devseunggwan) [reply docs/hook/INDEX.md]: Fixed in fa7add0. Clarified the row: `--model` compares a **bare Claude tier** only (silent for `codex:`/`gemini:`/full model IDs), scoped it to the complexity→model phase of the two-phase Provider Routing model, and moved the always-loaded block removal to the companion ai-dotfiles change (this PR ships only the mechanism). Same edit applied to ARCHITECTURE.md:180 and the spec intro.

File: docs/bypass-vars.md [hunk 2]
```diff
@@ -41,6 +41,7 @@ in [`../SECURITY.md`](../SECURITY.md).
 | `PRAXIS_MOMENTUM_BYPASS` | `momentum-rule-retrieval-gate` | Skip the high-momentum rule nudge |
 | `PRAXIS_VERSION_BUMP_BYPASS` | `version-bump-evidence-check` | Skip the version-bump evidence advisory |
 | `PRAXIS_SKIP_STAGED_FILE_ENUM` | `pre-commit-staged-file-enumeration` | Skip the pre-commit staged-file enumeration advisory |
+| `PRAXIS_SKIP_MODEL_ROUTING` | `model-routing-advisory` | Skip the delegation `--model` tier-mismatch advisory |
```
[the remark this answers is not in the document]
Comment (devseunggwan) [reply docs/bypass-vars.md:44]: Skipping — the row matches the file convention. The table header is "Effect when set" and every sibling `PRAXIS_SKIP_*` row omits the exact value while its impl uses the identical `== "1"` check (e.g. `PRAXIS_SKIP_STAGED_FILE_ENUM`). Documenting the `1` value on only this row would diverge from the file; a file-wide "set to 1" note is a separate cleanup out of scope for this PR.
